packages/protocol/migrations_sol/run_integration_tests_in_anvil.sh

```diff
@@ -1,13 +1,16 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-
-# generate devchain
+# Generate and run devchain
+echo "Generating and running devchain before running integration tests..."
 source $PWD/migrations_sol/create_and_migrate_anvil_devchain.sh
 
 # Run integration tests
-source $PWD/migrations_sol/integration_tests.sh
-
+echo "Running integration tests..."
+forge test \
+--match-path test-sol/integration/Integration.t.sol \
+-vvv \
+--fork-url http://127.0.0.1:$ANVIL_PORT
```

Comment:
Looks like the integration test passes locally but fails on CI:

```sh
    ├─ [0] VM::getDeployedCode("FeeCurrencyDirectory.sol") [staticcall]
    │   └─ ← [Revert] No matching artifact found
    └─ ← [Revert] No matching artifact found

Suite result: FAILED. 1 passed; 1 failed; 0 skipped; finished in 248.21ms (249.50ms CPU time)

Ran 1 test suite in 264.20ms (248.21ms CPU time): 1 tests passed, 1 failed, 0 skipped (2 total tests)

Failing tests:
Encountered 1 failing test in test-sol/integration/Integration.t.sol:RegistryIntegrationTest
[FAIL. Reason: No matching artifact found] test_shouldHaveCorrectBytecode() (gas: 33002125)
```

Source: https://github.com/celo-org/celo-monorepo/actions/runs/9320393687/job/25657108425?pr=11002#step:18:7060

Reply:
This is fixed in https://github.com/celo-org/celo-monorepo/pull/11002/commits/41bff1b8feefed6a019d2c840557665270c0fbc0

Reply:
When I exclude Integration tests, some other tests seem to fail: 

```
forge test -vvv --no-match-path "test-sol/integration/Integration.t.sol"
```

I'll look into this next.

Reply:
Looks like this PR https://github.com/celo-org/celo-monorepo/pull/10997
1. [removed the fixed Foundry version](https://github.com/celo-org/celo-monorepo/commit/bb2fd67b78c8a231edf0c2ddd28b19f58c9a9e59#diff-ca2c25d3008d874c6f407def69dff8c80817e9fe7536f9ab83d28ea24e7c637fL42-L43) on CI (so my local Foundry version is out of date)
2. [removed the Solidity version](https://github.com/celo-org/celo-monorepo/commit/bb2fd67b78c8a231edf0c2ddd28b19f58c9a9e59#diff-f1a37af096f1a0b4aedeecac6028b46624cd60bcd8a35b7332426f4b3b3d8f1bL16) from the contract name

Reply:
Upgrading my local foundry version fixed the `SECP256K1.sol` bug.

I was on 

```
forge --version
forge 0.2.0 (1a2e2e0 2023-11-14T00:27:18.137399000Z)
```

Upgraded with:

```
foundryup
```

Cleaned `out/` directory:

```sh
forge clean
```

Re-ran test:

```
forge test \
-vvvv \
--no-match-path "test-sol/integration/Integration.t.sol" \
--no-match-contract "RegistryIntegrationTest"
```

Reply:
The only failing test has this error </summary>

```
Failing tests:
Encountered 3 failing tests in contracts/common/libraries/test/BLS12Passthrough.sol:BLS12_377Passthrough
[FAIL. Reason: EvmError: InvalidFEOpcode; counterexample: calldata=0x9548643d0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000e4ece1fc6123a1ca06d777718f22e000000000000000000000000000000000000 args=[0x4ece1fc6123a1ca06d777718f22e]] testDeserialize(bytes) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x0001e0 with length 0x00000e. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x4de0f08d0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000e4ece1fc6123a1ca06d777718f22e000000000000000000000000000000000000 args=[0x4ece1fc6123a1ca06d777718f22e]] testParseG1(bytes) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x000360 with length 0x00000e. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x092176a70000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000e4ece1fc6123a1ca06d777718f22e000000000000000000000000000000000000 args=[0x4ece1fc6123a1ca06d777718f22e]] testParseG2(bytes) (runs: 0, μ: 0, ~: 0)

Encountered 3 failing tests in contracts/common/libraries/test/BLS12Passthrough.sol:BLS12_381Passthrough
[FAIL. Reason: EvmError: InvalidFEOpcode; counterexample: calldata=0x9548643d0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000e4ece1fc6123a1ca06d777718f22e000000000000000000000000000000000000 args=[0x4ece1fc6123a1ca06d777718f22e]] testDeserialize(bytes) (runs: 2, μ: 7796, ~: 7796)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x0001e0 with length 0x00000e. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x4de0f08d0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000e4ece1fc6123a1ca06d777718f22e000000000000000000000000000000000000 args=[0x4ece1fc6123a1ca06d777718f22e]] testParseG1(bytes) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x000360 with length 0x00000e. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x092176a70000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000e4ece1fc6123a1ca06d777718f22e000000000000000000000000000000000000 args=[0x4ece1fc6123a1ca06d777718f22e]] testParseG2(bytes) (runs: 0, μ: 0, ~: 0)

Encountered 3 failing tests in contracts/common/libraries/test/BLS12Passthrough.sol:Common
[FAIL. Reason: EvmError: InvalidFEOpcode; counterexample: calldata=0x9548643d0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000e4ece1fc6123a1ca06d777718f22e000000000000000000000000000000000000 args=[0x4ece1fc6123a1ca06d777718f22e]] testDeserialize(bytes) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x0001e0 with length 0x00000e. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x4de0f08d0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000e4ece1fc6123a1ca06d777718f22e000000000000000000000000000000000000 args=[0x4ece1fc6123a1ca06d777718f22e]] testParseG1(bytes) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x000360 with length 0x00000e. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x092176a70000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000e4ece1fc6123a1ca06d777718f22e000000000000000000000000000000000000 args=[0x4ece1fc6123a1ca06d777718f22e]] testParseG2(bytes) (runs: 0, μ: 0, ~: 0)

Encountered 1 failing test in contracts/identity/test/RandomTest.sol:RandomTest
[FAIL. Reason: revert: randomness should be zero if there is no previous commitment; counterexample: calldata=0x0c0dd6c178c84ece1fc6123a1ca06d777718f22edd3af9d970ec65aea47c8f195dda377ff483f25396ffda615fd8b51bad2e669f1bc9cc7bb51cbac1eab59b3ce427d783000000000000000000000000fdd29ce5b1e8214d5fb164cc7e4af1b8f7e08640 args=[0x78c84ece1fc6123a1ca06d777718f22edd3af9d970ec65aea47c8f195dda377f, 0xf483f25396ffda615fd8b51bad2e669f1bc9cc7bb51cbac1eab59b3ce427d783, 0xFDd29CE5b1E8214D5Fb164cC7E4aF1B8F7e08640]] testRevealAndCommit(bytes32,bytes32,address) (runs: 0, μ: 0, ~: 0)
```

Reply:
@pahor167 kindly offered to help fix the odd failing tests above.

Reply:
When I reproduce the contract calls with `chisel`, I don't get the same errors:

```sh
chisel
Welcome to Chisel! Type `!help` to show available commands.
➜ !fork http://127.0.0.1:8546
Set fork URL to http://127.0.0.1:8546
➜ bytes memory expectedBytecodeWithMetadata = vm.getDeployedCode("FeeCurrencyDirectory.sol");
➜ expectedBytecodeWithMetadata
Type: dynamic bytes
├ Hex (Memory):
├─ Length ([0x00:0x20]): 0x0000000000000000000000000000000000000000000000000000000000000af0
├─ Contents ([0x20:..]): 0x608060405234801561001057600080fd5b50600436106100b45760003560e01c8063715018a611610071578063715018a6146101905780638129fc1c146101985780638da5cb5b146101a0578063eab43d97146101c9578063efb7601d14610245578063f2fde38b1461026d57600080fd5b8063158ef93e146100b957806316be73a8146100db578063216ab7df146100f057806354255be0146101035780636036cba31461012957806361c661de1461017b575b600080fd5b6000546100c69060ff1681565b60405190151581526020015b60405180910390f35b6100ee6100e9366004610939565b610280565b005b6100ee6100fe366004610963565b61045d565b6001806000806040805194855260208501939093529183015260608201526080016100d2565b61015c61013736600461099f565b600160208190526000918252604090912080549101546001600160a01b039091169082565b604080516001600160a01b0390931683526020830191909152016100d2565b61018361062a565b6040516100d291906109c1565b6100ee61068c565b6100ee6106c8565b60005461010090046001600160a01b03166040516001600160a01b0390911681526020016100d2565b6102216101d736600461099f565b604080518082018252600080825260209182018190526001600160a01b03938416815260018083529083902083518085019094528054909416835292909201549181019190915290565b6040805182516001600160a01b0316815260209283015192810192909252016100d2565b61025861025336600461099f565b610731565b604080519283526020830191909152016100d2565b6100ee61027b36600461099f565b610823565b6000546001600160a01b036101009091041633146102b95760405162461bcd60e51b81526004016102b090610a0e565b60405180910390fd5b60025481106103005760405162461bcd60e51b8152602060048201526013602482015272496e646578206f7574206f6620626f756e647360681b60448201526064016102b0565b816001600160a01b03166002828154811061031d5761031d610a43565b6000918252602090912001546001600160a01b03161461037f5760405162461bcd60e51b815260206004820152601a60248201527f496e64657820646f6573206e6f74206d6174636820746f6b656e00000000000060448201526064016102b0565b6001600160a01b0382166000908152600160208190526040822080546001600160a01b03191681558101919091556002805490916103bc91610a59565b815481106103cc576103cc610a43565b600091825260209091200154600280546001600160a01b0390921691839081106103f8576103f8610a43565b9060005260206000200160006101000a8154816001600160a01b0302191690836001600160a01b03160217905550600280548061043757610437610a80565b600082815260209020810160001990810180546001600160a01b03191690550190555050565b6000546001600160a01b0361010090910416331461048d5760405162461bcd60e51b81526004016102b090610a0e565b6001600160a01b0382166104e35760405162461bcd60e51b815260206004820152601d60248201527f4f7261636c6520616464726573732063616e6e6f74206265207a65726f00000060448201526064016102b0565b600081116105335760405162461bcd60e51b815260206004820152601c60248201527f496e7472696e736963206761732063616e6e6f74206265207a65726f0000000060448201526064016102b0565b6001600160a01b0383811660009081526001602052604090205416156105a55760405162461bcd60e51b815260206004820152602160248201527f43757272656e637920616c726561647920696e20746865206469726563746f726044820152607960f81b60648201526084016102b0565b6040805180820182526001600160a01b039384168152602080820193845294841660008181526001968790529283209151825495166001600160a01b031995861617825592519085015560028054948501815590527f405787fa12a823e0f2b7631cc41b3ba8828b3321ca811111fa75cd3aa3bb5ace90920180549091169091179055565b6060600280548060200260200160405190810160405280929190818152602001828054801561068257602002820191906000526020600020905b81546001600160a01b03168152600190910190602001808311610664575b5050505050905090565b6000546001600160a01b036101009091041633146106bc5760405162461bcd60e51b81526004016102b090610a0e565b6106c660006108c4565b565b60005460ff161561071b5760405162461bcd60e51b815260206004820152601c60248201527f636f6e747261637420616c726561647920696e697469616c697a65640000000060448201526064016102b0565b6000805460ff191660011790556106c6336108c4565b6001600160a01b03818116600090815260016020526040812054909182911661079c5760405162461bcd60e51b815260206004820152601d60248201527f43757272656e6379206e6f7420696e20746865206469726563746f727900000060448201526064016102b0565b6001600160a01b038381166000818152600160205260409081902054905163efb7601d60e01b815260048101929092529091169063efb7601d906024016040805180830381865afa1580156107f5573d6000803e3d6000fd5b505050506040513d601f19601f820116820180604052508101906108199190610a96565b9094909350915050565b6000546001600160a01b036101009091041633146108535760405162461bcd60e51b81526004016102b090610a0e565b6001600160a01b0381166108b85760405162461bcd60e51b815260206004820152602660248201527f4f776e61626c653a206e6577206f776e657220697320746865207a65726f206160448201526564647265737360d01b60648201526084016102b0565b6108c1816108c4565b50565b600080546001600160a01b03838116610100818102610100600160a81b0319851617855560405193049190911692909183917f8be0079c531659141344cd1fd0a4f28419497f9722a3daafe3b4186f6b6457e091a35050565b80356001600160a01b038116811461093457600080fd5b919050565b6000806040838503121561094c57600080fd5b6109558361091d565b946020939093013593505050565b60008060006060848603121561097857600080fd5b6109818461091d565b925061098f6020850161091d565b9150604084013590509250925092565b6000602082840312156109b157600080fd5b6109ba8261091d565b9392505050565b6020808252825182820181905260009190848201906040850190845b81811015610a025783516001600160a01b0316835292840192918401916001016109dd565b50909695505050505050565b6020808252818101527f4f776e61626c653a2063616c6c6572206973206e6f7420746865206f776e6572604082015260600190565b634e487b7160e01b600052603260045260246000fd5b81810381811115610a7a57634e487b7160e01b600052601160045260246000fd5b92915050565b634e487b7160e01b600052603160045260246000fd5b60008060408385031215610aa957600080fd5b50508051602090910151909290915056fea26469706673582212209f369c037206599fbce507c5cc6cd46b4b1e30bd0db2c3a593da13e9c2fbcdc164736f6c6343000813003300000000000000000000000000000000
``` 

I'll keep digging into why this fails when we use `forge test`.

Reply:
Looks like this is a known issue in: 

> It seems that starting from the **Foundry** nightly build of **2024-05-02**, there were breaking changes in how pre-built contracts are managed during tests: https://github.com/foundry-rs/foundry/releases/tag/nightly-cafc2606a2187a42b236df4aa65f4e8cdfcea970
> 
> Until this problem is resolved, please use the version preceding these breaking changes to utilize the `foundry-chainlink-toolkit` and run its tests: https://github.com/foundry-rs/foundry/releases/tag/nightly-f625d0fa7c51e65b4bf1e8f7931cd1c6e2e285e9
> 
> To install this version in your environment, please run: `foundryup -v nightly-f625d0fa7c51e65b4bf1e8f7931cd1c6e2e285e9`

Source: https://github.com/smartcontractkit/foundry-chainlink-toolkit/issues/15#issuecomment-2104490631

I'm downgrading the version to check if that fixes the issue.

```
foundryup -v nightly-f625d0fa7c51e65b4bf1e8f7931cd1c6e2e285e9
```

Reply:
With the fiedf foundry version, the integration test passes ✅ 

```
forge test \
--match-path test-sol/integration/Integration.t.sol \
-vvv \
--fork-url http://127.0.0.1:8546

Suite result: ok. 2 passed; 0 failed; 0 skipped; finished in 269.86ms (269.78ms CPU time)

Ran 1 test suite in 277.51ms (269.86ms CPU time): 2 tests passed, 0 failed, 0 skipped (2 total tests)
```

I'll fix the foundry version in the CI workflow 

```
foundryup -v nightly-f625d0fa7c51e65b4bf1e8f7931cd1c6e2e285e9
```

Reply:
Left a comment in the PR that removed the fixed foundry version:

> (for future reference)
> 
> I ran into a limitation with this in this PR:
> 
> * [Integration tests for Anvil migrations #11002](https://github.com/celo-org/celo-monorepo/pull/11002)
> 
> The latest version has a regression that leads to a failing integration test. But, fixing a previous nightly build fixed the error.
> 
> More context [here](https://github.com/celo-org/celo-monorepo/pull/11002/files#r1624810719).

Source: https://github.com/celo-org/celo-monorepo/pull/10997/files#r1624826163

Reply:
With the fixed foundry version, the migration script passes ✅

```
./migrations_sol/run_integration_tests_in_anvil.sh

Suite result: ok. 2 passed; 0 failed; 0 skipped; finished in 255.18ms (254.56ms CPU time)

Ran 1 test suite in 272.34ms (255.18ms CPU time): 2 tests passed, 0 failed, 0 skipped (2 total tests)
Killing Anvil
```

Reply:
But, on that version the "file not found" error is back:

```
Failing tests:
Encountered 1 failing test in test-sol/governance/voting/ReleaseGold.t.sol:ReleaseGoldTest_AuthorizeWithPublicKeys
[FAIL. Reason: setup failed: failed to read from "/home/runner/work/celo-monorepo/celo-monorepo/packages/protocol/out/SECP256K1.sol/SECP256K1.json": No such file or directory (os error 2)] setUp() (gas: 0)

Encountered 1 failing test in test-sol/governance/voting/ReleaseGold.t.sol:ReleaseGoldTest_GetWithdrawableAmount
[FAIL. Reason: failed to read from "/home/runner/work/celo-monorepo/celo-monorepo/packages/protocol/out/SECP256K1.sol/SECP256K1.json": No such file or directory (os error 2)] test_ShouldReturnOnlyUpToItsOwnBalanceWhenValidatorSignerAuthorizedWithPublicKey() (gas: 77914)
```

Source: [CI](https://github.com/celo-org/celo-monorepo/actions/runs/9354744167/job/25748282476?pr=11002#step:13:530)

On the next nightly (`nightly-cafc2606a2187a42b236df4aa65f4e8cdfcea970`), build the weird `MemView` errors are introduced:

 ```
Failing tests:
Encountered 3 failing tests in contracts/common/libraries/test/BLS12Passthrough.sol:BLS12_377Passthrough
[FAIL. Reason: EvmError: InvalidFEOpcode; counterexample: calldata=0x9548643d00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000 args=[0x]] testDeserialize(bytes) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x0001e0 with length 0x000000. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x4de0f08d00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000 args=[0x]] testParseG1(bytes) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x000360 with length 0x000000. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x092176a700000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000 args=[0x]] testParseG2(bytes) (runs: 0, μ: 0, ~: 0)

Encountered 3 failing tests in contracts/common/libraries/test/BLS12Passthrough.sol:BLS12_381Passthrough
[FAIL. Reason: EvmError: InvalidFEOpcode; counterexample: calldata=0x9548643d00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000 args=[0x]] testDeserialize(bytes) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x0001e0 with length 0x000000. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x4de0f08d00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000 args=[0x]] testParseG1(bytes) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x000360 with length 0x000000. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x092176a700000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000 args=[0x]] testParseG2(bytes) (runs: 0, μ: 0, ~: 0)

Encountered 3 failing tests in contracts/common/libraries/test/BLS12Passthrough.sol:Common
[FAIL. Reason: EvmError: InvalidFEOpcode; counterexample: calldata=0x9548643d00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000 args=[0x]] testDeserialize(bytes) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x0001e0 with length 0x000000. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x4de0f08d00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000 args=[0x]] testParseG1(bytes) (runs: 0, μ: 0, ~: 0)
[FAIL. Reason: revert: TypedMemView/index - Overran the view. Slice is at 0x000360 with length 0x000000. Attempted to index at offset 0x000000 with length 0x000020.; counterexample: calldata=0x092176a700000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000 args=[0x]] testParseG2(bytes) (runs: 0, μ: 0, ~: 0)

Encountered 1 failing test in contracts/identity/test/RandomTest.sol:RandomTest
[FAIL. Reason: revert: slicing out of range; counterexample: calldata=0x0c0dd6c1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 args=[0x0000000000000000000000000000000000000000000000000000000000000000, 0x0000000000000000000000000000000000000000000000000000000000000000, 0x0000000000000000000000000000000000000000]] testRevealAndCommit(bytes32,bytes32,address) (runs: 0, μ: 0, ~: 0)

Encountered a total of 10 failing tests, 2078 tests succeeded
```

I think I'll stick to the nightly version (`nightly-f625d0fa7c51e65b4bf1e8f7931cd1c6e2e285e9`), and will revert the diff that introduced the "file not found" error:

```diff
+ deployCodeTo("out/SECP256K1.sol/SECP256K1.0.5.17.json", SECP256K1Address);
- deployCodeTo("SECP256K1.sol:SECP256K1", SECP256K1Address);
```

Source: https://github.com/celo-org/celo-monorepo/pull/10997

Reply:
I just noticed the problematic tests (`RandomTest` and `BLS12Passthrough.sol`) are explicitly excluded from the foundry configs:

https://github.com/celo-org/celo-monorepo/blob/1f7ab42e7fda270f1a05da8e42560fa875fed4ee/packages/protocol/foundry.toml#L20-L22

When I exclude integration tests with `--no-match-path` and `--no-match-contract `, it probably overrides the `foundry.toml` config above, which leads to the failing tests:

```
forge test \
-vvv \
--no-match-path "test-sol/integration/Integration.t.sol" \
--no-match-contract "RegistryIntegrationTest"
```

Reply:
When I exclude these files manually, the tests pass:

```sh
forge test \
-vvv \
--no-match-contract "RegistryIntegrationTest|RandomTest|BLS12_381Passthrough"
```

`--no-match-path` cannot be used multiple time like:

```
--no-match-path "test-sol/integration/Integration.t.sol" \
--no-match-path "contracts/common/libraries/test/BLS12Passthrough.sol" \
```

Instead, `--no-match-contract` can take a regex-like argument to exclude multiple test files like

```
--no-match-contract "RegistryIntegrationTest|RandomTest|BLS12_381Passthrough"
```

Reply:
### Foundry versions with the regression

I fixed the version at [`nightly-f625d0fa7c51e65b4bf1e8f7931cd1c6e2e285e9`](https://github.com/foundry-rs/foundry/releases/tag/nightly-f625d0fa7c51e65b4bf1e8f7931cd1c6e2e285e9) (2024-04-02), because it's the last foundry version that doesn't have the regression:

1. [`nightly-fbad377ab26a432e48444cf324feee1195a30960`](https://github.com/foundry-rs/foundry/releases/tag/nightly-fbad377ab26a432e48444cf324feee1195a30960) (2024-06-04) has the regression ❌
1. [`nightly-0fc39763d46b8ffab5fa4eaeb2f65ae078fa07de`](https://github.com/foundry-rs/foundry/releases/tag/nightly-0fc39763d46b8ffab5fa4eaeb2f65ae078fa07de) (2024-06-03) has the regression ❌ 
1. [`nightly-5ac78a9cd4b94dc53d1fe5e0f42372b28b5a7559`](https://github.com/foundry-rs/foundry/releases/tag/nightly-5ac78a9cd4b94dc53d1fe5e0f42372b28b5a7559) (2024-06-02) has the regression ❌
1. [`nightly-f479e945c6be78bb902df12f9d683c3bb55e3fb0`](https://github.com/foundry-rs/foundry/releases/tag/nightly-f479e945c6be78bb902df12f9d683c3bb55e3fb0) (2024-06-01) has the regression ❌ 
1. [`nightly-cafc2606a2187a42b236df4aa65f4e8cdfcea970`](https://github.com/foundry-rs/foundry/releases/tag/nightly-cafc2606a2187a42b236df4aa65f4e8cdfcea970) (2024-05-02) has the regression ❌ 
1. [`nightly-f625d0fa7c51e65b4bf1e8f7931cd1c6e2e285e9`](https://github.com/foundry-rs/foundry/releases/tag/nightly-f625d0fa7c51e65b4bf1e8f7931cd1c6e2e285e9) (2024-04-02) doesn't have the regression ✅ 

### Context on the regression

> In the meantime, I'm fixing a failing test in `Integration.t.sol` that is due to upgrading the foundry version:
> 
> ```
> ├─ [0] console::log("Checking bytecode of:", "FeeCurrencyDirectory") [staticcall]
>     │   └─ ← [Stop] 
>     ├─ [3888] 0x000000000000000000000000000000000000ce10::getAddressForStringOrDie("FeeCurrencyDirectory") [staticcall]
>     │   ├─ [3048] 0xF09E8DCA7ca021a11CC5927fC8fDD3b701Acf87e::getAddressForStringOrDie("FeeCurrencyDirectory") [delegatecall]
>     │   │   └─ ← [Return] 0x42Fe5a2A61ed9705eb2F08a04A58CEB606D22f6a
>     │   └─ ← [Return] 0x42Fe5a2A61ed9705eb2F08a04A58CEB606D22f6a
>     ├─ [2406] 0x42Fe5a2A61ed9705eb2F08a04A58CEB606D22f6a::_getImplementation() [staticcall]
>     │   └─ ← [Return] 0x63837e6E68C9070DF4f1CD70DCdcce4644b42b0E
>     ├─ [0] VM::getDeployedCode("FeeCurrencyDirectory.sol") [staticcall]
>     │   └─ ← [Revert] No matching artifact found
>     └─ ← [Revert] No matching artifact found
> ```
> 
> Source: [CI](https://github.com/celo-org/celo-monorepo/actions/runs/9353463199/job/25744020440#step:18:4488)
> 
> On my local foundry version, this test passed, but since I upgraded my local foundry version, the test fails.
> 
> I have a hunch it has to do with contract versions, because `FeeCurrencyDirectory` is a 0.8 contract, whereas most other contracts are 0.5.
> 
> I'll look int this hypothesis now.

> With the fixed foundry version, the integration test passes ✅
> 
> ```
> forge test \
> --match-path test-sol/integration/Integration.t.sol \
> -vvv \
> --fork-url http://127.0.0.1:8546
> 
> Suite result: ok. 2 passed; 0 failed; 0 skipped; finished in 269.86ms (269.78ms CPU time)
> 
> Ran 1 test suite in 277.51ms (269.86ms CPU time): 2 tests passed, 0 failed, 0 skipped (2 total tests)
> ```
> 
> I fixed the foundry version in [1f7ab42](https://github.com/celo-org/celo-monorepo/commit/1f7ab42e7fda270f1a05da8e42560fa875fed4ee)
> 
> ```
> foundryup -v nightly-f625d0fa7c51e65b4bf1e8f7931cd1c6e2e285e9
> ```

Reply:
As of https://github.com/celo-org/celo-monorepo/pull/11002/commits/2f9f8379ae80d5f1a076a5290c936e1bc0c0477c, the integration tests are excluded correctly and all tests pass as expected ✅